### ISSUE TITLE
Creating existing register with different owner does not give error

### ISSUE THREAD

**Issue by b-zee** [maidsafe/safe_network#836]
Using `Client::store_public_register` twice providing different owners succeeds.

```
let mut csprng = rand::rngs::OsRng;
let pk1: Keypair = Keypair::new_ed25519(&mut csprng).public_key();
let pk2: Keypair = Keypair::new_ed25519(&mut csprng).public_key();

let xor = XorName::random();

let (_, wal) = client
    .store_public_register(xor, 0xFF, pk1, Default::default())
    .await
    .unwrap();
client.publish_register_ops(wal).await.unwrap();

let (_, wal) = client
    .store_public_register(xor, 0xFF, pk2, Default::default())
    .await
    .unwrap();
client.publish_register_ops(wal).await.unwrap(); // No panic ...
```

Is this expected behavior?

**Comment by bochaco** [comment]
No it's not expected/correct, this may be solved as part of changes of PR #929 (cc: @maqi )

**Comment by bochaco** [comment]
I just tested this with `main` branch of safe_network (commit `155d6225`) and the issue persists, I'll try to work on a fix.
